src/platform/Darwin/DnssdImpl.cpp

```diff
@@ -215,20 +275,55 @@ static void OnGetAddrInfo(DNSServiceRef sdRef, DNSServiceFlags flags, uint32_t i
 {
     ChipLogProgress(Discovery, "Mdns: %s flags: %d, interface: %u, hostname: %s", __func__, flags, (unsigned) interfaceId,
                     StringOrNullMarker(hostname));
-    auto sdCtx = reinterpret_cast<ResolveContext *>(context);
+
+    auto contextWithType = reinterpret_cast<ResolveContextWithType *>(context);
+    VerifyOrReturn(contextWithType != nullptr, ChipLogError(Discovery, "ResolveContextWithType is null"));
+
+    auto sdCtx = reinterpret_cast<ResolveContext *>(contextWithType->context);
     ReturnOnFailure(MdnsContexts::GetInstance().Has(sdCtx));
     LogOnFailure(__func__, err);
 
     if (kDNSServiceErr_NoError == err)
     {
-        sdCtx->OnNewAddress(interfaceId, address);
+        InterfaceKey interfaceKey = { interfaceId, hostname, contextWithType->isSRPType };
+        sdCtx->OnNewAddress(interfaceKey, address);
+
+        // Set the flag to start the timer for resolve on SRP domain to complete if the key has the SRP type requested flag set to
+        // true.
+        if (interfaceKey.isSRPTypeRequested)
+        {
+            sdCtx->shoulStartSRPTimerForResolve = true;
```

Comment:
This logic is making no sense to me.  interfaceKey.isSRPTypeRequested will be true if we have already gotten an IP address from some SRP thing (that's how we get into this function with this flag set).  Why would we do any timer-starting in that case?

Reply:
I fixed the timer logic based on what I think I understood by re-visiting the notes I took. I messed up before. For now the mental model is when calling ResolveWithContext based on whether we start a resolve on the SRP domain. In OnGetAddrInfo after successfully adding a new address if the address resolved was on SRP domain, i set shouldStartSRPTimerForResolve to false. hope that's correct.